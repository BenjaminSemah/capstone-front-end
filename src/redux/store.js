import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import courseReducer from './coursesSlice';
import { userReducer } from './auth';
import reservationSlice from './reservationSlice';

const store = configureStore({
  reducer: {
    UserReducer: userReducer,
    course: courseReducer,
    reservation: reservationSlice,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
