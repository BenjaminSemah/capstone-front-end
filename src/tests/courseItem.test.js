import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import CourseItem from '../components/courseItem';

describe('Courses', () => {
  it('renders correctly', () => {
    const tree = render(<Provider store={store}><Router><CourseItem /></Router></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
