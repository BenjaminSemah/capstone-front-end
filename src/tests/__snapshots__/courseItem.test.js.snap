// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Courses renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-center col-md-4 col-12"
      >
        <a
          href="/details/undefined"
          style="text-decoration: none; color: rgb(151, 191, 15);"
        >
          <img
            class="rounded-circle home-img"
            draggable="false"
          />
          <div>
            <p
              class="fw-bold"
            />
            <p />
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-center col-md-4 col-12"
    >
      <a
        href="/details/undefined"
        style="text-decoration: none; color: rgb(151, 191, 15);"
      >
        <img
          class="rounded-circle home-img"
          draggable="false"
        />
        <div>
          <p
            class="fw-bold"
          />
          <p />
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
